src/app/tests/suites/certification/Test_TC_OO_2_4.yaml

```diff
@@ -238,6 +238,21 @@ tests:
       PICS: OO.S.C00.Rsp
       command: "Off"
 
+      # We get an acknowledgment that the command off above is recieved, but it takes
+      # a few milliseconds for the off command to actually turn it off. Without this
+      # delay the preceding reboot command might happen before the command is fully
```

Comment:
Why is there that delay?  The server should be responding after it's already processed the command...

This seems to be papering over what might be a bug in the on/off server implementation.  We should figure out what exactly is going on here and whether it's OK.

Reply:
This has something to do with the level-control calling `scheduleTimerCallbackMs` in `src/app/clusters/level-control/level-control.cpp` I think this is by design I didn't want to yak-shave further then this, but I can if you really think there is a bug here

Reply:
@jmartinez-silabs @Rob-Houtepen-Signify Is replying with success to the Off command before the OnOff attribute has actually been set to false correct?

Reply:
The command handling is done directly when the command is received and if everything is valid, it will respond success.

Now, for the actual state of the OnOff state. There is a level transition to be done from the current level we are at to the minimum allowed for the device/endpoint (with LT featuremap set). 

Full context:
There were many discussions and arguments last summer about when the OnOff state should be done. Initially, it was directly when the command is received even if the level transition was ongoing. It was left to the application to deal with that and it was done like this because the OnOff cluster spec says 

```
7.1. Off Command
This command does not have any data fields.

7.1.1. Effect on Receipt
On receipt of the Off command, a server SHALL set the OnOff attribute to FALSE.
```

This led to arguing and discussion about the above statement and how it should interact with level control. It was finally agreed, to facilitate application logic, with the lighting feature set and level-control combo, the on-off OFF state must be set once the level transition is complete. I didn't implement the change but did approve the PR.

It does make sense from an application point of view. However, That change brought a lot of corner case issues with the interaction of the different clusters and they are hard to preemptively identify. The test plan/test run should be able to validate all that logic. I have been fixing the issues as they arise.

Now I believe the small delay seen here is to be expected with the current implementation. as the actual off-state change is done from a timer callback linked with the level transition.

If you feel this is problematic and the command success response should only be done at the end of the transition. File a bug and I'll address it.
Do you have an opinion on what should be the behaviour of the response @Rob-Houtepen-Signify

Reply:
I have opened up https://github.com/CHIP-Specifications/connectedhomeip-spec/issues/6329

This is to track:
```
the question is whether the response should be going out immediately or after the transition is done.
Especially in cases like this, where I believe the transition is "instantaneous" (as in, a single step)....

This should really be defined in the spec, ideally.
```